### PR TITLE
refactor: change order of params for Server

### PR DESCRIPTION
**Pull request by rishabh3112**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
Will update tests accordingly

### Motivation / Use-Case
We are moving to plugin interface for the server. The ideal signature for that would be `constructor(options={})` but we need compiler in params as well for backward compatibility. For that reason, this PR aims to change the order of params so that plugin implementation can ignore compiler param.

### Breaking Changes
Yes, we will need to fix this in serve package after patch release here

### Additional Info
https://github.com/webpack/webpack-dev-server/pull/3251/files#r629851714

### CODE REVIEW

**Review by anshumanv — DISMISSED** [review]
Thanks!! 🌟

**Comment by rishabh3112** [comment]
> Thanks!! :star2:

Hey! No need to thank as such but appreciated 😄.

File: lib/Server.js [hunk 1]
```diff
@@ -39,7 +39,12 @@ if (!process.env.WEBPACK_SERVE) {
 }
 
 class Server {
-  constructor(compiler, options = {}) {
+  constructor(options = {}, compiler) {
+    // TODO: remove this after plugin support is published
+    if (options.hooks) {
```
Comment (alexander-akait) [inline lib/Server.js:44]: Let's rename `options.plugin`, but will not docs it, it is internal

Comment (rishabh3112) [reply lib/Server.js:44]: I didn't get what you want to say here 😅

Comment (alexander-akait) [reply lib/Server.js:44]: Oh I see, you just check `options.hooks` to if it is exists, it is compiler, otherwise options, right?

Comment (rishabh3112) [reply lib/Server.js:44]: Yup

Comment (alexander-akait) [reply lib/Server.js:44]: Good idea

**Review by alexander-akait — DISMISSED** [review]

**Review by alexander-akait — COMMENTED** [review]
Oh, can we test multi compiler mode too or we already have tests?

**Comment by rishabh3112** [comment]
> Oh, can we test multi compiler mode too or we already have tests?

Nothing is changed in this PR for multi-compilers though, will add similar test for multi-compiler too anyway.

**Review by snitin315 — APPROVED** [review]
LGTM

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
Thanks

**Comment by alexander-akait** [comment]
Now we can continue development dev server as plugin and don't worry about breaking change for v4

**Pull request merged**